### PR TITLE
added "Merged" in searchissues which allows search repos by merged da…

### PR DESCRIPTION
**Pull request by chenjiaming93**
…te with existing syntax.

it generates a CA1502 code excessive complexity warning and i suppressed it.
#839

### CODE REVIEW

File: Octokit/Models/Request/SearchIssuesRequest.cs [hunk 1]
```diff
@@ -43,7 +43,7 @@ public SearchIssuesRequest(string term, string owner, string name)
         }
 
         /// <summary>
-        /// Optional Sort field. One of comments, created, or updated. 
+        /// Optional Sort field. One of comments, created, updated,or merged
```
Comment (Haacked) [inline Octokit/Models/Request/SearchIssuesRequest.cs]: Minor nit but there should be a space after "updated," there. :smile:

File: Octokit.Tests/Clients/SearchClientTests.cs [hunk 2]
```diff
@@ -1045,7 +1045,78 @@ public void TestingTheCreatedQualifier_Between()
                     Arg.Is<Uri>(u => u.ToString() == "search/issues"),
                     Arg.Is<Dictionary<string, string>>(d => d["q"] == "something+created:2014-01-01..2014-02-02"));
             }
+            [Fact]
```
Comment (shiftkey) [inline Octokit.Tests/Clients/SearchClientTests.cs]: :lipstick: insert a newline between the `{` and the `[Fact]` to give each test some space

File: Octokit/Models/Request/SearchIssuesRequest.cs (hunk 1 above)
Comment (shiftkey) [inline Octokit/Models/Request/SearchIssuesRequest.cs]: :thumbsup:

**Comment by shiftkey** [comment]
@chenjiaming93 a couple of little things to tidy up, but this is looking great!

File: Octokit.Tests/Clients/SearchClientTests.cs [hunk 3]
```diff
@@ -1045,7 +1045,78 @@ public void TestingTheCreatedQualifier_Between()
...
+                var request = new SearchIssuesRequest("something");
+                request.Merged = DateRange.LessThan(new DateTime(2014, 1, 1));
+
+                client.SearchIssues(request);
+
+                connection.Received().Get<SearchIssuesResult>(
+                    Arg.Is<Uri>(u => u.ToString() == "search/issues"),
+                    Arg.Is<Dictionary<string, string>>(d => d["q"] == "something+merged:<2014-01-01"));
+            }
+            [Fact]
```
Comment (shiftkey) [inline Octokit.Tests/Clients/SearchClientTests.cs]: And here

File: Octokit.Tests/Clients/SearchClientTests.cs [hunk 4]
```diff
@@ -1045,7 +1045,78 @@ public void TestingTheCreatedQualifier_Between()
...
+                var request = new SearchIssuesRequest("something");
+                request.Merged = DateRange.LessThanOrEquals(new DateTime(2014, 1, 1));
+
+                client.SearchIssues(request);
 
+                connection.Received().Get<SearchIssuesResult>(
+                    Arg.Is<Uri>(u => u.ToString() == "search/issues"),
+                    Arg.Is<Dictionary<string, string>>(d => d["q"] == "something+merged:<=2014-01-01"));
+            }
+            [Fact]
```
Comment (shiftkey) [inline Octokit.Tests/Clients/SearchClientTests.cs]: And here

File: Octokit.Tests/Clients/SearchClientTests.cs [hunk 5]
```diff
@@ -1045,7 +1045,78 @@ public void TestingTheCreatedQualifier_Between()
...
+                var client = new SearchClient(connection);
+                var request = new SearchIssuesRequest("something");
+                request.Merged = DateRange.Between(new DateTime(2014, 1, 1), new DateTime(2014, 2, 2));
+
+                client.SearchIssues(request);
+
+                connection.Received().Get<SearchIssuesResult>(
+                    Arg.Is<Uri>(u => u.ToString() == "search/issues"),
+                    Arg.Is<Dictionary<string, string>>(d => d["q"] == "something+merged:2014-01-01..2014-02-02"));
+            }
```
Comment (shiftkey) [inline Octokit.Tests/Clients/SearchClientTests.cs]: And after this line

**Comment by shiftkey** [comment]
Closing in favour of #957

**Pull request closed without merging**